src/Controller/AddressController.php

```diff
@@ -87,6 +88,70 @@ public function index(Request $request, String $hash): Response
         ]);
     }
 
+    /**
+     * @Route("/address/cold/{hash}")
+     */
+    public function cold(Request $request, String $hash): Response
+    {
+        $data = [];
+
+        $types = ["Staking", "Spending", "Voting"];
+
+        try {
+            if (strlen($hash) == 61) {
+                $coldStaking = "V1";
+            } elseif (strlen($hash) == 89) {
+                $coldStaking = "V2";
+            }
+
+            $version = 53;
+            $coldStakingV1 = 21;
+            $coldStakingV2 = 36;
+
+            if ($this->getParameter('navcoin.network') != 'mainnet') {
+                $version = 111;
+                $coldStakingV1 = 8;
+                $coldStakingV2 = 32;
+            }
+
+            $coldStakingVersion = $coldStakingV1;
+            if ($coldStaking == "V2") {
+                $coldStakingVersion = $coldStakingV2;
+            }
+
+            $base58 = new Base58([
+                "characters" => Base58::BITCOIN,
+                "check" => true,
+                "version" => $coldStakingVersion
+            ]);
+
+            $base58_nav = new Base58([
+                "characters" => Base58::BITCOIN,
+                "check" => true,
+                "version" => $version
+            ]);
+
+            $decoded = $base58->decode($hash);
+
+            $addresses = str_split($decoded, 20);
+
+            for ($i = 0; $i < count($addresses); $i++) {
+                $address = $base58_nav->encode($addresses[$i]);
+                $data[$i] =  new \stdClass;
+                $data[$i]->address = $this->addressApi->getAddress($address);
+                $data[$i]->summary = $this->summaryApi->getSummary($address);
```

Comment:
These will call I think 6 API requests on Cold Staking V2 addresses and 4 API request on ColdStaking V1 addresses.